File: result/metagpt/claude/humaneval/90.c
#include <stdio.h>
#include <stdlib.h>

int next_smallest(int* lst, int size) {
    if (size <= 1) return -1; // 如果列表为空或只有一个元素，返回-1表示没有第二小的元素

    // 对数组进行排序
    for (int i = 0; i < size - 1; i++) {
        for (int j = i + 1; j < size; j++) {
            if (lst[i] > lst[j]) {
                int temp = lst[i];
                lst[i] = lst[j];
                lst[j] = temp;
            }
        }
    }

    // 找到第二小的元素
    for (int i = 1; i < size; i++) {
        if (lst[i] != lst[i - 1]) {
            return lst[i];
        }
    }

    return -1; // 如果没有第二小的元素，返回-1
}

File: result/metagpt/deepseek/humaneval/90.rs
pub fn next_smallest(mut lst: Vec<i32>) -> Option<i32> {
    if lst.len() <= 1 {
        return None;
    }
    
    lst.sort_unstable();
    
    for i in 1..lst.len() {
        if lst[i] != lst[i - 1] {
            return Some(lst[i]);
        }
    }
    
    None
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_next_smallest() {
        assert_eq!(next_smallest(vec![1, 2, 3]), Some(2));
        assert_eq!(next_smallest(vec![1, 1, 1]), None);
        assert_eq!(next_smallest(vec![1]), None);
    }
}